src/zenml/steps/entrypoint_function_utils.py

```diff
@@ -235,17 +236,40 @@ def _validate_input_value(
             parameter: The function parameter for which the value was provided.
             value: The input value.
         """
-        config_dict = ConfigDict(arbitrary_types_allowed=False)
-
-        # Create a pydantic model with just a single required field with the
-        # type annotation of the parameter to verify the input type including
-        # pydantics type coercion
-        validation_model_class = create_model(
-            "input_validation_model",
-            __config__=config_dict,
-            value=(parameter.annotation, ...),
-        )
-        validation_model_class(value=value)
+        annotation = parameter.annotation
+
+        # Handle Optional types
+        origin = get_origin(annotation)
+        if origin is Union:
+            args = get_args(annotation)
+            if type(None) in args:
+                if value is None:
+                    return  # None is valid for Optional types
+                # Remove NoneType from args as this case is handled from here
+                args = tuple(arg for arg in args if arg is not type(None))
+                annotation = args[0] if len(args) == 1 else Union[args]
+
+        # Handle None values for non-Optional types
+        if value is None and annotation is not type(None):
```

Comment:
What is this for? This would mean someone would explicitly have a `a: type(None)` annotation right, or how else would we get into this if condition?